src/libcore/tests/mem.rs

```diff
@@ -134,12 +134,4 @@ fn test_discriminant_send_sync() {
 fn test_const_forget() {
     const _: () = forget(0i32);
     const _: () = forget(Vec::<Vec<Box<i32>>>::new());
-
-    // Writing this function signature without const-forget
-    // triggers compiler errors:
-    // 1) That we use a non-const fn inside a const fn
-    // 2) without the forget, it complains about the destructor of Box
```

Comment:
I found these quite good reasons to keep the method around. Just add `#[allow(unused)]` to it, and maybe `// FIXME this method cannot be called in const-eval yet`.